presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/PrestoUninstaller.java

```diff
@@ -83,4 +89,18 @@ private static void uninstallUsingRpm(boolean checkDependencies, boolean forceUn
         }
         LOGGER.debug("Successfully uninstalled Presto");
     }
+
+    public static void uninstallUsingTar()
+            throws PrestoManagerException
+    {
+        // TODO: Stop Presto before uninstall
+        try {
+            deleteRecursively(TAR_INSTALLATION_DIR);
```

Comment:
This may cause security problem with recursive deletion if any symbolic link exists. You would like to catch this `InsecureRecursiveDeleteException` and handle it separately. Ditto for the deleteDirectoryContents method you had earlier. This applies to all the places you use this method.